solutions/src/main/java/ru/sberbank/school/task13/cacheproxy/serialization/SerializationManager.java

```diff
@@ -0,0 +1,66 @@
+package ru.sberbank.school.task13.cacheproxy.serialization;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.KryoException;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import ru.sberbank.school.task13.cacheproxy.exception.CacheProxyIOException;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class SerializationManager {
+    private final Kryo kryo;
+
+    public SerializationManager() {
+        this.kryo = new Kryo();
+        this.kryo.register(String.class);
+        this.kryo.register(Object.class);
+        this.kryo.register(Class.class);
+        this.kryo.register(ArrayList.class);
+    }
+
+    public void writeMethodInFile(File file, Object result) throws CacheProxyIOException {
+        try (Output output = new Output(new FileOutputStream(file))) {
+            if (result != null) {
+                kryo.register(result.getClass());
+            }
+            kryo.writeClassAndObject(output, result);
+        } catch (KryoException | FileNotFoundException e) {
+            throw new CacheProxyIOException(e.getMessage());
+        }
+    }
+
+    public Object readMethodFromFile(File file) throws CacheProxyIOException {
+        try (Input input = new Input(new FileInputStream(file))) {
+            return kryo.readClassAndObject(input);
+        } catch (FileNotFoundException e) {
+            throw new CacheProxyIOException(e.getMessage());
+        }
+    }
+
+    public void writeInZip(File file, Object result) throws CacheProxyIOException {
+        try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(file + ".zip"));
+             Output output = new Output(zos)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a141275970>*